integration_test/common/common.go

```diff
@@ -104,30 +107,47 @@ func SliceContains(slice []string, toFind string) bool {
 
 func NewIntegrationMetadataValidator() *validator.Validate {
 	v := validator.New()
-	_ = v.RegisterValidation("onerepresentative", func(fl validator.FieldLevel) bool {
+	_ = v.RegisterValidation("onetrue", func(fl validator.FieldLevel) bool {
+		field := fl.Field()
+		param := fl.Param()
 
-		fieldStr := fl.Field().String()
-		if fieldStr == "" {
-			// Ignore the case where this field is not actually specified or is left empty.
-			return true
+		if param == "" {
+			panic(fmt.Sprintf("onetrue must contain an argument"))
 		}
 
-		metrics, ok := fl.Field().Interface().([]*ExpectedMetric)
-		if !ok {
-			return false
-		}
+		switch field.Kind() {
 
-		if len(metrics) == 0 {
-			return true
-		}
+		case reflect.Slice, reflect.Array:
+			elem := field.Type().Elem()
```

Comment:
Where is this used? Were you planning to check that the resulting type is a struct that has the requested field?